File: DataAccessLayer/Entities/WaterPoint.cs
using System;
using System.Collections.Generic;
using System.ComponentModel.DataAnnotations;
using System.ComponentModel.DataAnnotations.Schema;
using System.Diagnostics.CodeAnalysis;
using System.Linq;
using System.Text;
using System.Threading.Tasks;

namespace DataAccessLayer.Entities
{
    public class WaterPoint: BaseEntity
    {
        [Required, StringLength(50)]
        public string Title { get; set; }

        [Required, StringLength(200)]
        public string Description { get; set; }

        [Required]
        public WaterPointType WaterPointType { get; set; }

        [Required, StringLength(20)]
        public string PhoneNumber { get; set; }

        [Required, StringLength(80)]
        public string Address { get; set; }

        [Required, Range(0d,5d)]
        public double Rating { get; set; }

        [Required]
        public bool IsVerified { get; set; }

        [Required]
        public bool HasOrdering { get; set; }

        [Required]
        public bool HasOwnDelivery { get; set; }

        [Required]
        public bool HasSearchPriority { get; set; }

        [StringLength(50)]
        public string photoUrl { get; set; }

        [Required, DatabaseGenerated(DatabaseGeneratedOption.None)]
        public int UserId { get; set; }

        [Required, DatabaseGenerated(DatabaseGeneratedOption.None), AllowNull]
        public int? VerificationDocumentId { get; set; }

        public VerificationDocument VerificationDocument { get; set; }

        public RegisteredUser User { get; set; }

        public List<Review> Reviews { get; set; }

        public List<Product> Products { get; set; }


    }

    public enum WaterPointType
    {
        PumpStation,
        WaterDelivery,
        Shop,
        EmergencyWaterTransportation,
        Other
    }
}

File: UI-Angular/src/models/water-point.ts


export class WaterPoint{

    id!: number;
    title: string;
    description: string;
    waterPointType: WaterPointType;
    phoneNumber: string;
    address: string;
    rating: number;
    isVerified: boolean;
    hasOrdering: boolean;
    hasOwnDelivery: boolean;
    hasSearchPriority: boolean;
    userId: number;
    verificationDocumentId!: number;
    photoUrl: string;

    reviewIds: Array<number>;
    productIds: Array<number>;

    constructor(title: string, description: string, waterPointType: WaterPointType, 
    phoneNumber: string, address: string, rating: number, isVerified: boolean, 
    hasOrdering: boolean, hasOwnDelivery: boolean, hasSearchPriority: boolean, 
    userId: number, verificationDocumentId: number, 
    reviewIds: Array<number>, productIds: Array<number>, photoUrl: string){
        this.title = title;
        this.description = description;
        this.waterPointType = waterPointType;
        this.phoneNumber = phoneNumber;
        this.address = address;
        this.rating = rating;
        this.isVerified = isVerified;
        this.hasOrdering = hasOrdering;
        this.hasOwnDelivery = hasOwnDelivery;
        this.userId = userId;
        this.hasSearchPriority = hasSearchPriority;
        this.reviewIds = reviewIds;
        this.productIds = productIds;
        this.photoUrl = photoUrl;
        this.verificationDocumentId = verificationDocumentId;
    }
}

export enum WaterPointType{
    PumpStation,
    WaterDelivery,
    Shop,
    EmergencyWaterTransportation,
    Other
}